src/convertToObject.js

```diff
@@ -5,8 +5,16 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  let arr = sourceString.split(';');
+  arr = arr.filter((rule) => rule.trim().length);
+  for (const line of arr) {
+    const [rule, value] = line.split(':');
+    result[rule.trim()] = value.trim();
+  }
```

Comment:
[CODE PRACTICE] - The task checklist advises to avoid using loops and instead use iteration methods. Consider using the `forEach` method, which is an exception to the rule against modifying an existing array/object, as it doesn't return a new array but allows you to apply a function to each element of the array.